sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/ViewModelSuspendingFunctionsCheck.kt

```diff
@@ -0,0 +1,35 @@
+package org.sonarsource.kotlin.checks
+
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.psiUtil.isPublic
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.BindingContext.FUNCTION
+import org.jetbrains.kotlin.resolve.descriptorUtil.fqNameOrNull
+import org.jetbrains.kotlin.resolve.descriptorUtil.getAllSuperClassifiers
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.AbstractCheck
+import org.sonarsource.kotlin.api.suspendModifier
+import org.sonarsource.kotlin.plugin.KotlinFileContext
+
+@Rule(key = "S6313")
+class ViewModelSuspendingFunctionsCheck : AbstractCheck() {
+
+    override fun visitNamedFunction(function: KtNamedFunction, kotlinFileContext: KotlinFileContext) {
+        val bindingContext = kotlinFileContext.bindingContext
+        if (function.isPublic
+            && function.suspendModifier() != null
+            && function.extendsViewModel(bindingContext)
+        ) {
+            kotlinFileContext.reportIssue(function.nameIdentifier!!,
```

Comment:
We are reporting on the fact that the dev is making a function suspending here - do you think it could be valuable to highlight the `suspend` modifier?